### DISCUSSION TITLE
Optional Savestate and Savegame upload for Report Broken Achievement

### DISCUSSION THREAD

**Discussion by ghost** [RetroAchievements/RAWeb#712] [Ideas]
A Savestate and Savegame upload function for the **Report Broken Achievement** page.

- **Savestate**
 for simple Achievements (mostly without hits), right before the Achievement should appear but doesn't gets triggered or gets triggered at wrong time.

- **Savegame**
 for more complex Achievements where accuracy matters.

> **Tip:**  Recommended to upload both, Savestate right before the situation and Savegame from last Save Point to be more accuracte.

**Additional functions**
- Auto-deleting inactive Savefiles that are older than x days/weeks/months.
- A function to keep Savefiles forever (preventing from being Auto-deleted).
- A function to manually delete a Savefile by a Developer when the Achievement issue has been solved.
- A Savefile Description/Comment function by Developers for future usage.

**Comment by Jamiras** [comment]
It seems like these ideas are better suited for the toolkit. RAWeb could simply provide a way to attach files to the ticket.

> A Savefile Description/Comment function by Developers for future usage.
* Assuming the file is attached to a ticket, the comments could go there.

As for automatically capturing this information:

> gets triggered at wrong time
* This is easy enough.  We know when the achievement triggers. But the save state would be as the achievement triggers, not right before. I'd hate to recommend capturing every frame just so we have the frame prior to an error as the overhead would be substantial. And the longer between captured frames, the less likely the exact situation to reproduce the problem might occur.

> right before the Achievement should appear
* How do we determine this generically? It might be possible to always capture the state if an achievement is one condition away from triggering, but some achievements trigger off multiple conditions firing at the same time (value = X, delta = Y is the simplest case). Also see my concern about the overhead of capturing frames unnecessarily.

> Savegame
* This is good idea for games that have them. It'll at least get the developer close to the problem. And they're typically smaller files than save states.

**Comment by julkasalsa** [comment]
What about valuable server space? Some of  those files, especially states are quite heavy. Won't they potentially bring stress to server traffic, especially acummulated?

**Comment by kdecks** [comment]
Related https://github.com/RetroAchievements/RAIntegration/issues/187
